test/e2etest/crowd_test.go

```diff
@@ -0,0 +1,262 @@
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+	formData := url.Values{
+		"name":                                {"Crowd " + directoryName},
+		"crowdServerUrl":                      {crowdURL},
+		"applicationName":                     {"crowd-openid-server"},
+		"applicationPassword":                 {crowdPassword},
+		"crowdPermissionOption":               {"READ_ONLY"},
+		"_nestedGroupsEnabled":                {"visible"},
+		"incrementalSyncEnabled":              {"true"},
+		"_incrementalSyncEnabled":             {"visible"},
+		"groupSyncOnAuthMode":                 {"ALWAYS"},
+		"crowdServerSynchroniseIntervalInMin": {"1"},
+		"save":                                {"Save and Test"},
+		"atl_token":                           {atlToken},
+		"directoryId":                         {"0"},
+	}
+	encodedData := formData.Encode()
+	request, err := http.NewRequest(http.MethodPost, bitbucketURL+"/plugins/servlet/embedded-crowd/configure/crowd/", strings.NewReader(encodedData))
+	request.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	cookie := &http.Cookie{Name: "BITBUCKETSESSIONID", Value: bitbucketSessionID}
+	request.AddCookie(cookie)
+	_, err = client.Do(request)
+	require.NoError(t, err)
+}
+
+func getBitbucketSessionID(bitbucketURL string, username string, password string) (sessionID string) {
+	client := &http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+	formData := url.Values{
+		"j_username":      {username},
+		"j_password":      {password},
+		"atl_remember_me": {"on"},
+		"submit":          {"Log in"},
+	}
+	encodedData := formData.Encode()
+	request, err := http.NewRequest(http.MethodPost, bitbucketURL+"/j_atl_security_check", strings.NewReader(encodedData))
+	request.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	resp, err := client.Do(request)
+	if err != nil {
+		fmt.Println(err)
+	}
+	cookies := resp.Cookies()
+	var bitbucketSessionID *http.Cookie
+	for _, cookie := range cookies {
+		if cookie.Name == "BITBUCKETSESSIONID" {
+			bitbucketSessionID = cookie
+			return bitbucketSessionID.Value
+		}
+	}
+	return ""
+}
+
+func generateCrowdCfgXml(t *testing.T, testConfig TestConfig, jdbcURL string, rdsPassword string) {
+
+	vars := make(map[string]interface{})
+	vars["crowd_rds_password"] = rdsPassword
+	vars["crowd_jdbc_url"] = jdbcURL
+	vars["crowd_license"] = testConfig.CrowdLicense
+
+	// parse the template
+	tmpl, _ := template.ParseFiles("crowd-backup/crowd.cfg.xml.tpl")
+
+	// create a new file
+	file, _ := os.Create("crowd.cfg.xml")
+	defer file.Close()
+
+	// apply the template to the vars map and write the result to file.
+	err := tmpl.Execute(file, vars)
+	require.NoError(t, err, "Error applying template to crowd.cfg.xml file")
+}
+
+func restoreCrowdFromBackup(t *testing.T, testConfig TestConfig, jdbcURL string) {
```

Comment:
Rather than have ebs and rds backups in all regions, for e2e tests it's easier to restore DB using sql dump, then process crowd.cfg.xml, copy it to shared home and restart crowd pod.